analyzers/src/SonarAnalyzer.VisualBasic/Rules/LinkedListPropertiesInsteadOfMethods.cs

```diff
@@ -0,0 +1,40 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules.VisualBasic;
+
+[DiagnosticAnalyzer(LanguageNames.VisualBasic)]
+public sealed class LinkedListPropertiesInsteadOfMethods : LinkedListPropertiesInsteadOfMethodsBase<SyntaxKind, InvocationExpressionSyntax>
+{
+    protected override ILanguageFacade<SyntaxKind> Language => VisualBasicFacade.Instance;
+
+    protected override bool TryGetOperands(InvocationExpressionSyntax invocation, out SyntaxNode left, out SyntaxNode right) =>
+        invocation.TryGetOperands(out left, out right);
+
+    protected override bool HasAnyArguments(InvocationExpressionSyntax invocation) => false;
```

Comment:
I am not a big fan of this. I understand the logic, but it feels like the abstraction is not "abstracting" anymore, but gets in the way of readability and hurts more than it helps. 

If I came up to this code after a while, I would be wondering why this always returns false.
The same goes for the `invocation` argument below on line 33, it is useless for the C# implementation.

I think you should refactor the logic in the derived classes to solve these two problems. 
Personally, I prefer slight duplication and more readability, than less code and variables/methods that are not used.

Reply:
I wasn't a big fan either, I took some time to refactor the logic, please take a look and let me know what you think